### PR TITLE
prefer popup over preview window

### PR DESCRIPTION
**Pull request by Konfekt**
addresses #1092

### CODE REVIEW

File: ftplugin/python/jedi.vim [hunk 1]
```diff
@@ -44,10 +44,15 @@ if g:jedi#auto_initialization
         inoremap <silent> <buffer> <space> <C-R>=jedi#smart_auto_mappings()<CR>
     end
 
-    if g:jedi#auto_close_doc
+    if g:jedi#auto_close_doc && (&g:completeopt =~# '\<preview\>' && &g:completeopt !~# '\<popup\>')
         " close preview if its still open after insert
         augroup jedi_preview
-            autocmd! InsertLeave <buffer> if pumvisible() == 0|pclose|endif
+            if v:version > 704
```
Comment (davidhalter) [inline ftplugin/python/jedi.vim:50]: What's the reasoning for this change?

**Comment by davidhalter** [comment]
I like the changes a lot, I'm just unsure why you changed that one specific thing.

File: ftplugin/python/jedi.vim (hunk 1 above)
Comment (Konfekt) [reply ftplugin/python/jedi.vim:50]: Hello, 

I just implemented [the first two answers](https://stackoverflow.com/questions/3105307/how-do-you-automatically-remove-the-preview-window-after-autocompletion-in-vim) as `:help g:jedi#auto_close_doc` states that

> this window is being closed after insertion of a completion item.

the `CompleteDone` event seemed the more exact autocmd to achieve this, whenever available.

Comment (davidhalter) [reply ftplugin/python/jedi.vim:50]: Thanks that makes sense.

**Comment by davidhalter** [comment]
The coverage tests are currently failing, I feel like this is something unrelated. Maybe @blueyed has any idea? I feel like if he does not chime in I might disable the coverage tests temporarily.

**Comment by davidhalter** [comment]
Sorry forgot to merge, because I spilled tea on my notebook and it took me a while to get it back in good shape. Now that I have it back, I found the tab of this PR again. Thanks a lot again :)

**Pull request merged**